cato/client.go

```diff
@@ -0,0 +1,332 @@
+	data, err := json.Marshal(obj)
+	if err != nil {
+		return nil
+	}
+
+	var mapRepresentation map[string]interface{}
+	err = json.Unmarshal(data, &mapRepresentation)
+	if err != nil {
+		a.conf.ClientOptions.DebugLog(err.Error())
+		return nil
+	}
+
+	return mapRepresentation
+}
+
+func (a *CatoAdapter) handleEvent(marker string, account_id string, api_key string) uintptr {
+
+	start = time.Now()
+
+	eventFilterString := ""
+	eventSubfilterString := ""
+
+	iteration := 1
+	totalCount := 0
+
+	for {
+		query := fmt.Sprintf(`{
+  eventsFeed(accountIDs:[%q]
+    marker:%q
+    filters:[%s,%s])
+  {
+    marker
+    fetchedCount
+    accounts {
+      id
+      records {
+        time
+        fieldsMap
+      }
+    }
+  }
+}`, account_id, marker, eventFilterString, eventSubfilterString)
+
+		success, resp := a.send(query, account_id, api_key)
+		if !success {
+			a.conf.ClientOptions.DebugLog(fmt.Sprintf("%s", resp))
+			break
+		}
+
+		marker = resp["data"].(map[string]interface{})["eventsFeed"].(map[string]interface{})["marker"].(string)
+		fetchedCount := int(resp["data"].(map[string]interface{})["eventsFeed"].(map[string]interface{})["fetchedCount"].(float64))
+		totalCount += fetchedCount
+
+		line := fmt.Sprintf("iteration:%d fetched:%d total_count:%d marker:%s", iteration, fetchedCount, totalCount, marker)
+
+		records := resp["data"].(map[string]interface{})["eventsFeed"].(map[string]interface{})["accounts"].([]interface{})[0].(map[string]interface{})["records"].([]interface{})
+		if len(records) > 0 {
+			line += " " + records[0].(map[string]interface{})["time"].(string)
+			line += " " + records[len(records)-1].(map[string]interface{})["time"].(string)
+		}
+
+		eventsList := make([]map[string]interface{}, 0)
+		for _, event := range records {
+			fieldsMap := event.(map[string]interface{})["fieldsMap"].(map[string]interface{})
+			fieldsMap["event_timestamp"] = event.(map[string]interface{})["time"].(string)
+			eventReorder := make(map[string]interface{})
+			for k, v := range fieldsMap {
+				if k == "event_timestamp" {
+					eventReorder[k] = v
+					break
+				}
+			}
+			for k, v := range fieldsMap {
+				if k != "event_timestamp" {
+					eventReorder[k] = v
+				}
+			}
+			eventsList = append(eventsList, eventReorder)
+		}
+
+		for _, event := range eventsList {
+
+			msg := &protocol.DataMessage{
+				JsonPayload: a.convertStructToMap(event),
+				TimestampMs: uint64(time.Now().UnixNano() / int64(time.Millisecond)),
+			}
+			err := a.uspClient.Ship(msg, a.writeTimeout)
+			if err == uspclient.ErrorBufferFull {
+				a.conf.ClientOptions.OnWarning("stream falling behind")
+				err = a.uspClient.Ship(msg, 0)
+			}
+			if err != nil {
+				a.conf.ClientOptions.OnError(fmt.Errorf("Ship(): %v", err))
+			}
+		}
+
+		if err := ioutil.WriteFile(configFile, []byte(marker), 0644); err != nil {
+			a.conf.ClientOptions.DebugLog(fmt.Sprintf("Error writing config file: %v", err))
+		}
+
+		iteration++
+	}
+
```

Comment:
You will need a return here if breaking out, or can dropped the required return type from this method.